src/AggregateFunctions/Streaming/CountedValueMap.h

```diff
@@ -1,52 +1,85 @@
 #pragma once
 
+#include <base/StringRef.h>
+#include <Common/ArenaUtils.h>
+#include <Common/ArenaWithFreeLists.h>
+
 #include <absl/container/btree_map.h>
 
+#include <map>
 #include <stdexcept>
 
 namespace DB
 {
 namespace Streaming
 {
+template <typename T>
+struct CountedValueArena
+{
+    CountedValueArena() = default;
+    T emplace(T key) { return key; }
```

Comment:
If `T` is a string or a complex type, `arena->emplace(std::move(v))` will construct string again by using move ctor  and `emplace(T key)` will return the newly construct key which will be used to construct the key in the Map. So there is a new string ctor cost which may be trivial.

Can we do it better by ? Please experiment it out.

`T emplace(T && key) { return key; }`